go/libkb/secret_store.go

```diff
@@ -246,3 +248,74 @@ func (s *SecretStoreLocked) GetUsersWithStoredSecrets(m MetaContext) ([]string,
 	}
 	return s.disk.GetUsersWithStoredSecrets(m)
 }
+
+func (s *SecretStoreLocked) PrimeSecretStores(mctx MetaContext) (err error) {
+	if s == nil || s.isNil() {
+		return errors.New("secret store is not available")
+	}
+	if s.disk != nil {
+		err = PrimeSecretStore(mctx, s.disk)
+		if err != nil {
+			return err
+		}
+	}
+	err = PrimeSecretStore(mctx, s.mem)
+	return err
+}
+
+// PrimeSecretStore runs a test with current platform's secret store, trying to
+// store, retrieve, and then delete a secret with an arbitrary name. This should
+// be done before provisioning or logging in
+func PrimeSecretStore(mctx MetaContext, ss SecretStoreAll) (err error) {
+	defer mctx.TraceTimed("PrimeSecretStore", func() error { return err })()
+
+	// Generate test username and test secret
+	testUsername, err := RandString("test_ss_", 5)
+	testUsername = strings.Replace(testUsername, "=", "_", -1)
+	if err != nil {
+		return err
+	}
+	randBytes, err := RandBytes(LKSecLen)
+	if err != nil {
+		return err
+	}
+	mctx.Debug("PrimeSecretStore: priming secret store with username %q and secret %v", testUsername, randBytes)
+	testNormUsername := NormalizedUsername(testUsername)
+	var secretF [LKSecLen]byte
+	copy(secretF[:], randBytes[:])
+	testSecret := LKSecFullSecret{f: &secretF}
+
+	// Try to fetch first, we should get an error back.
+	_, err = ss.RetrieveSecret(mctx, testNormUsername)
+	if err == nil {
+		return errors.New("managed to retrieve secret before storing it")
+	} else if err != nil {
+		mctx.Debug("PrimeSecretStore: error when retrieving secret that wasn't stored yet: %q, as expected", err)
+	}
+
+	// Put secret in secret store through `SecretStore` interface.
+	err = ss.StoreSecret(mctx, testNormUsername, testSecret)
+	if err != nil {
+		return err
+	}
+	defer func() {
```

Comment:
I think you should move this up in the function, it's possible the `StoreSecret` returns an error but there is data written that we want to clear